src/main/java/io/cryostat/core/util/RuleFilterParser.java

```diff
@@ -48,24 +48,36 @@
 import org.apache.commons.lang3.StringUtils;
 
 public class RuleFilterParser {
-    private static final Set<String> RULE_IDS_SET =
-            RuleRegistry.getRules().stream().map(rule -> rule.getId()).collect(Collectors.toSet());
+    private final Set<String> RULE_IDS_SET;
+    private final Set<String> TOPIC_IDS_SET;
```

Comment:
Nice. Now that theses are instance members rather than `static`, the naming convention should be ex. `topicIds` not `TOPIC_IDS`. I would take the `set` out of the name, it's redundant info since that's already the type declaration.